share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+
+4. Create a file with `.wake` extension and copy the functions mentioned in this tutorial and run them using `wake` command in the terminal.  
+
+5. For more information on wake library click [here](https://sifive.github.io/wake/)
+
+### wake `Hello World`
+* If you want to display a string you can do that using the below function by writing the string to be displayed inside "" and call the function abc . Ex: `wake -x 'abc'`. This will call the function `abc` and prints `Hello World`. 
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
+A function may be defined using the keyword `def`. Simple function definition takes the form
+`def <function name> <parameter list> = <expression>`
+
+* Below function  double takes an argment x and gives the output by multiplying the x by 2
+```
+def double x = 2*x
+```
+
+* Below function triple takes an argument x and gives the output by multiplying the x by 3
+```
+def triple x = 3*x
+```
+
+* Function inc increments the values by 1 and returns the value
+```
+def inc x = x+1
+```
+
+* Function adda takes an argument of type string and concatenates it with b and gives the output. `{ }` is the string interpolation operator
+```
+def adda s = "{s}b"
+```
+
+* Function times4 takes an argument x and applies the function double twice on x
+```
+def times4 x = double (double x)
+```
+
+* Function aveI(averageOfInteger) takes 2 arguments x and y and adds two values and divide the result by 2. This function shows the other was of defining the parameters using lambda operator. The traditional way of defining the parameter is `def aveI x y = (x+y)/2`. Either of them are logically same.
+```
+def aveI = \x\y (x+y)/2
+```
+
+* Function aveR(averageOfReal) takes 2 arguments of type double and divides the result by 2
+```
+def aveR = \x\y (x +. y) /. 2.0
+```
+
+* Function aveM(Modulus) adds 2 values and provides the remainder 
+```
+def aveM = \x\y (x+y) % 2
+```
+
+* Function duplicate concantenates or copies a given string twice
+```
+def duplicate s  ="{s}{s}"
+```
+
+* integerToUnicode is a `wake built-in function` which takes the ASCII input and provides its equivalent character
+```
+wake -x 'integerToUnicode 99'
+```
+
+* unicodeToInteger is a `wake built-in function` which gives the ASCII value of the given character.
+```
+wake -x 'unicodeToInteger "a"'
+```
+
+* strlen is a `wake built-in function` which returns the length of the given string.
+```
+wake -x 'strlen "abcde"'
+```
+
+* `|`is the pipeFn which takes the argument from the left hand side of the `|` and provides it to the right hand side function. This makes the function to be more readable. `(a(b(c(d))))` is represented as `d | c | b | a`. In both the cases d is evaluated first then followed by c , b and a.
+
+### Wake Built-In Functions
+Few of the commonly used built-in functions in this tutorial is given below
```

Comment:
can you break these up into operations on strings and operations on lists? For example, `explode` seems to be string specific. Is `cat` only for strings, or can you `cat` two lists of something that aren't strings?

Reply:
I can do that. 
Yes `explode` is string specific.
`cat` works only for strings